webapp/tests/karma/unit/services/enketo.js

```diff
@@ -968,7 +968,7 @@ describe('Enketo service', () => {
       dbGetAttachment.resolves('<form/>');
       UserContact.resolves({ _id: 'my-user', phone: '8989' });
       dbBulkDocs.callsFake(docs => Promise.resolve([ { ok: true, id: docs[0]._id, rev: '1-abc' } ]));
-      return service.save('my-form-internal-id', form, true).then(() => {
+      return service.save('my-form-internal-id', form, () => Promise.resolve(true)).then(() => {
```

Comment:
Let's update these tests to actually resolve the geo and ensure it's written to the docs. Some new tests for the geoError cases wouldn't hurt either!

Reply:
I've made a few of these slightly less terrible :+1:

Reply:
(keeping unresolved, waiting for decision on err logic before writing tests for that logic)